packages/flow-core/src/components/f-text/f-text.scss

```diff
@@ -201,9 +204,73 @@ f-div[direction="row"] {
 }
 
 :host {
+	@include base();
 	mark {
 		background-color: transparent;
 		color: var(--color-highlight-default);
 		display: contents;
 	}
+	.f-text-editable {
+		display: flex;
+		flex-direction: row;
+		width: 100%;
+	}
+	.pseudo-edit-text {
+		display: none;
+		position: absolute;
+		top: 0;
+		right: 0;
+	}
+	.pseudo-submit-text {
+		position: absolute;
+		top: 5px;
+		right: 6px;
+	}
+	.non-editable-slot {
+		width: 95%;
```

Comment:
you can use a calc function to calculate the exact width, subtract icos width from 100% `calc(100% - yourIconSize)`